q2_types/feature_data/_format.py

```diff
@@ -221,29 +168,123 @@ class PairedDNASequencesDirectoryFormat(model.DirectoryFormat):
 
 
 class AlignedDNAFASTAFormat(model.TextFileFormat):
-    def sniff(self):
-        filepath = str(self)
-        sniffer = skbio.io.io_registry.get_sniffer('fasta')
-        if sniffer(filepath)[0]:
-            generator = skbio.io.read(filepath, constructor=skbio.DNA,
-                                      format='fasta', verify=False)
-            try:
-                initial_length = len(next(generator))
-                for seq, _ in zip(generator, range(4)):
-                    if len(seq) != initial_length:
-                        return False
-                return True
-            # ValueError raised by skbio if there are invalid DNA chars.
-            except (StopIteration, ValueError):
-                pass
-        return False
+    def _validate_lines(self, max_lines):
+        FASTADNAValidator = re.compile(r'[ACGTURYKMSWBDHVN.-]+\r?\n?')
+        ValidationSet = frozenset(('A', 'C', 'G', 'T', 'U', 'R', 'Y', 'K', 'M',
+                                   'S', 'W', 'B', 'D', 'H', 'V', 'N', '.',
+                                   '-'))
+
+        _validate_DNAFASTAFormats(self, FASTADNAValidator, ValidationSet,
+                                  max_lines, True)
+
+    def _validate_(self, max_lines):
+        level_map = {'min': 100, 'max': float('inf')}
+        self._validate_lines(level_map[max_lines])
 
 
 AlignedDNASequencesDirectoryFormat = model.SingleFileDirectoryFormat(
     'AlignedDNASequencesDirectoryFormat', 'aligned-dna-sequences.fasta',
     AlignedDNAFASTAFormat)
 
 
+def _validate_DNAFASTAFormats(file, FASTADNAValidator, ValidationSet,
+                              max_lines, aligned=False):
+    last_line_was_ID = False
+    ids = {}
+
+    seq_len = 0
+    prev_seq_len = 0
+    prev_seq_start_line = 0
+
+    with open(str(file), 'rb') as fh:
```

Comment:
Since file is a pathlib object, you should be able to write `with file.open('rb') as fh:` (or something along those lines).

Reply:
The open method for `file` comes from `TextFileFormat` and looks like this 

```class TextFileFormat(_FileFormat):
    def open(self):
        mode = 'r' if self._mode == 'r' else 'r+'
        # ignore BOM only when reading, do not emit BOM on write
        encoding = 'utf-8-sig' if mode == 'r' else 'utf-8'
        return self.path.open(mode=mode, encoding=encoding)
```

I don't think it will let me do what I want, will it?

Reply:
`with file.path.open('rb') as fh:`